plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -5101,14 +5209,32 @@ private void fillHostNetworkInfo(VmwareContext serviceContext, StartupRoutingCom
         }
     }
 
-    private void fillHostDetailsInfo(VmwareContext serviceContext, Map<String, String> details) throws Exception {
+    private void fillHostDetailsInfo(final Map<String, String> details) throws Exception {
         VmwareHypervisorHost hyperHost = getHyperHost(getServiceContext());
 
         if (hyperHost.isHAEnabled()) {
             details.put("NativeHA", "true");
         }
     }
 
+    private void fillHostGpuInfo(final StartupRoutingCommand cmd) throws Exception {
+        final VmwareContext context = getServiceContext();
+        final VmwareHypervisorHost hyperHost = getHyperHost(context);
+        final HostMO hostMo = new HostMO(context, hyperHost.getMor());
+
+        try {
+            final HashMap<String, HashMap<String, VgpuTypesInfo>> groupDetails = hostMo.getGPUGroupDetails();
+            cmd.setGpuGroupDetails(groupDetails);
+            if (groupDetails != null && !groupDetails.isEmpty()) {
+                cmd.setHostTags("GPU");
+            }
+        } catch (final Exception e) {
```

Comment:
Can't we use a more specific exception here?

Reply:
Sure. I will modify this.